plugins/backend/catalog-backend-module-okta/src/providers/OktaOrgEntityProvider.ts

```diff
@@ -124,14 +126,22 @@ export class OktaOrgEntityProvider extends OktaEntityProvider {
         const defaultAnnotations = await this.buildDefaultAnnotations();
 
         await client.listUsers({ search: account.userFilter }).each(user => {
-          resources.push(
-            userEntityFromOktaUser(user, this.userNamingStrategy, {
-              annotations: defaultAnnotations,
-            }),
-          );
+          try {
+            const userName = this.userNamingStrategy(user);
+            userResources[userName] = userEntityFromOktaUser(
+              user,
+              this.userNamingStrategy,
+              {
+                annotations: defaultAnnotations,
+              },
+            );
+          } catch (e: unknown) {
+            assertError(e);
```

Comment:
Are you sure you want to use this here - this will throw an exception and exit the function just because you can't read the message property to log it, where it seems you want the error to be swallowed.

Reply:
You are breaking the execution of the rest of the code - is that what you want to do? Its preferable to me to log something even if the error message isn't there than log nothing and throw an exception up the stack that just says that the shape of something was not an error.. Does that make sense?

Reply:
We could/should implement an error log function that takes an error as the standard console.error does as a second arg, and then deals with doing the best job it can at logging something useful, whether thats stringifying the object or accessing the message and stack trace.

Reply:
i.e. <img width="366" alt="Screenshot 2023-03-31 at 16 47 57" src="https://user-images.githubusercontent.com/7934833/229153744-11d6af46-0998-4df7-9093-b0ac8c03ab5d.png">

Reply:
Throwing a generic type exception from code that is supposed to improve error handling seems like throwing the baby out with the bathwater to me